lib/confluence/set_ops.es6.js

```diff
@@ -158,13 +158,136 @@ foam.CLASS({
 });
 
 foam.CLASS({
+  package: 'org.chromium.dao',
+  name: 'UnionDAO',
+  extends: 'foam.dao.AbstractDAO',
+  implements: ['foam.mlang.Expressions'],
+
+  documentation: 'Read-only DAO that captures the union of two delegate DAOs.',
+
+  requires: ['foam.dao.ArraySink'],
+
+  classes: [
+    {
+      name: 'NoEOFSink',
+      extends: 'foam.dao.ProxySink',
+
+      methods: [function eof() {}],
+    },
+  ],
+
+  properties: [
+    {
+      class: 'foam.dao.DAOProperty',
+      name: 'primary',
+      required: true,
+    },
+    {
+      class: 'foam.dao.DAOProperty',
+      name: 'secondary',
+      required: true,
+    },
+    {
+      name: 'of',
+      factory: function() {
+        return this.primary.of;
+      },
+    },
+  ],
+
+  methods: [
+    function put(o) {
+      throw new Error('Attempt to put() to read-only UnionDAO');
+    },
+    function remove(o) {
+      throw new Error('Attempt to remove() from read-only UnionDAO');
+    },
+    function removeAll(o) {
+      throw new Error('Attempt to removeAll() from read-only UnionDAO');
+    },
+    function find(key) {
+      return new Promise((resolve, reject) => {
+        var data = {done: false, resolve: resolve, reject: reject};
+        var onFind = this.onFind.bind(this, data);
+        var onError = this.onFindError.bind(this, data);
+        this.primary.find(key).then(onFind, onError);
+        this.secondary.find(key).then(onFind, onError);
+      });
+    },
+    function select(opt_sink, skip, limit, order, predicate) {
+      var sink = opt_sink || this.ArraySink.create();
+      if (skip || limit || order || predicate)
+        throw new Error('UnionDAO does not support skip/limit/order/predicate');
+
+      return new Promise((resolve, reject) => {
+        var data = {doneCount: 0, errorCount: 0, sink: sink, resolve: resolve, reject: reject};
+        var onSelect = this.onSelect.bind(this, data);
+        var onError = this.onSelectError.bind(this, data);
+
+        // Select primary -> "sink"-with-no-eof() callback.
+        this.primary.select(this.NoEOFSink.create({delegate: sink}))
+            .then(onSelect, onError);
+        // Select secondary \ primary -> "sink"-with-no-eof() callback.
+        this.secondary.select(
+            this.SET_MINUS(this.primary, this.NoEOFSink.create({
+              delegate: sink,
+            })))
+                .then(onSelect, onError);
```

Comment:
This indentation feels weird. I think you want to either tack it to the end of the select, or indent +4 from the start of the statement.